### ISSUE TITLE
Sporadic drops in numbers with -alloc_space option

### ISSUE THREAD

**Issue by sangelovic** [gperftools/gperftools#1388]
After analyzing memory profiling data with `pprof` tool of two consecutive measurement files, I can sometimes observe drops in total numbers of allocated objects (or space) with -alloc_objects (or -alloc_space, respectively) pprof option. Do I understand that option correctly in that it counts all allocations (even those there were deallocated in the meantime)? Thus, the number shall be always increasing with every consecutive dump, right?

Still, in a dump file at a time "A" I can see for example:

```
flat  flat%   sum%        cum   cum%
1107  2.58% 79.81%       5122 11.96%  CBnPropertyContents::CBnPropertyContents
```

In other consecutive dumps this number is increasing, which is OK.

But at some point in a dump file generated at a time "X" > "A" I can see this:

```
flat  flat%   sum%        cum   cum%
688   1.20% 81.29%       4727  8.26%  CBnPropertyContents::CBnPropertyContents
```

I would expect that number to be always increasing. Is this a bug? Or did I get that option wrong? Thanks a lot for help.

**Comment by alk** [comment]
Thanks. This is likely fixed here: 44eb0ee83c3a485cf6af6366d06430fd162d0fa9

**Comment by sangelovic** [comment]
Thanks for looking into this and for heads up. We tried latest release (v2.13) today, and still there are drops, even though very sporadic ones:

0022.dump:

```
Showing nodes accounting for 32210.93kB, 94.81% of 33973.22kB total
...
         0     0% 94.81%   549.95kB  1.62%  CBnPropertyContents::Allocate
```

0023.dump:

```
Showing nodes accounting for 29660.17kB, 93.63% of 31679.30kB total
...
         0     0% 93.63%   542.27kB  1.71%  CBnPropertyContents::Allocate
```

The total numbers, as well as the number for `CBnPropertyContents::Allocate`, have decreased.

Is there still a bug in gperftools code?
